example/lib/samples/module.dart

```diff
@@ -0,0 +1,100 @@
+import 'package:example/samples/module_arguments.dart';
+import 'package:example/samples/modules/composer/module.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:nuvigator/next.dart';
+import 'package:provider/provider.dart';
+
+import 'bloc/samples_bloc.dart';
+import 'modules/friend_request/bloc/friend_request_bloc.dart';
+import 'modules/friend_request/module.dart';
+import 'screens/home_screen.dart';
+
+part 'module.g.dart';
+
+class HomeRoute extends NuRoute {
+  @override
+  Future<bool> init(BuildContext context) {
+    return Future.delayed(const Duration(seconds: 2), () => true);
+  }
+
+  @override
+  String get path => 'home';
+
+  @override
+  ScreenType get screenType => materialScreenType;
+
+  @override
+  Widget build(BuildContext context, NuRouteMatch<Object> match) {
+    return HomeScreen();
+  }
+
+  @override
+  Object parseParameters(Map<String, dynamic> map) {
+    return null;
+  }
+}
+
+@ModuleArguments()
```

Comment:
Any suggestion on the annotation name to generate the parser function?

Reply:
I would go with something more generic, I think that annotating the module with `ModuleArguments` is a bit strange.
Maybe `NuModuleGenerate`? or `@nuModule`, `@NuModuleParser`

Reply:
Thanks for the suggestions, I particularly find the @NuModuleParser great for this usecase.